File: 2021/fsharp/Aoc2021.Tests/Day07.Tests.fs
module Day07.Tests

open Xunit

let input = [ 16; 1; 2; 0; 4; 2; 7; 1; 2; 14 ]

[<Fact>]
let ``Day07 simulate works`` () =
    Assert.Equal((2, 37), input |> findBestPosition)
    Assert.Equal((5, 168), input |> findBestCrabPosition)

[<Fact>]
let ``Day07 Stars`` () =
    Assert.Equal("348996", job1 ())
    Assert.Equal("98231647", job2 ())

File: 2021/csharp/Aoc2021.Tests/Day07Tests.cs
namespace Aoc2021.Tests;

public class Day07Tests
{
    private static readonly List<int> input = "16,1,2,0,4,2,7,1,2,14"
        .Split(',').Select(Utils.ParseInt).ToList();

    [Fact]
    public void FindPositionWorks()
    {
        Assert.Equal((2, 37), Day07.FindBestPostion(input));
        Assert.Equal((5, 168), Day07.FindBestCrabPostion(input));
    }

    [Fact]
    public void Stars()
    {
        Day07 run = new();
        Assert.Equal("348996", run.Result1());
        Assert.Equal("98231647", run.Result2());
    }
}
